gm4_mountaineering/data/gm4_mountaineering/functions/player_main.mcfunction

```diff
@@ -0,0 +1,22 @@
+# Submain function to limit `as @a at @s` in main
+# @s = player not in spectator
+# at @s
+# run from main
+
+# Run ski-related functions
+execute at @s if entity @s[predicate=gm4_mountaineering:wearing_skis] run function gm4_mountaineering:ski_effects/skis_equipped
+
+# Clear ski-related functions when skis are removed
+execute at @s if entity @s[predicate=!gm4_mountaineering:wearing_skis,tag=gm4_mountaineering_using_skis] run function gm4_mountaineering:ski_effects/skis_removed
+
+# Adds jump boost effects when holding poles
+execute at @s if entity @s[predicate=gm4_mountaineering:holding_poles] run function gm4_mountaineering:pole_effects/poles_equipped
+
+# Initiate crampons when they are first equipped with a tag
+execute at @s if entity @s[tag=!gm4_mountaineering_using_crampons,predicate=gm4_mountaineering:wearing_crampons] run tag @s add gm4_mountaineering_using_crampons
+
+# Clear crampon tag when crampons are removed
+execute at @s if entity @s[predicate=!gm4_mountaineering:wearing_crampons,tag=gm4_mountaineering_using_crampons] run tag @s remove gm4_mountaineering_using_crampons
+
+# Apply damage to any climbing shulkers
+execute as @e[type=minecraft:shulker,tag=gm4_mountaineering_climb_assist] at @s run function gm4_mountaineering:climbing_effects/damage_shulkers
```

Comment:
I didn't think it really fit anywhere else (since the climbing functions are generally all run from the faster clock, but this doesn't need to be so isn't)
re: distance check, I'm not too sure how to implement this. I tried putting in `[distance=2..]` inside the `@e` call and following `@s` (which I thought would make sense) but that doesn't work. I'm also not sure how I'd make it unique to the player that made it - for example, shulkers and being used by one player shouldn't be getting damaged because another player is on the same server (and greater than 2 blocks away). So for now, I just have it that all shulkers get damaged and killed off regardless of how a player uses them (justified by saying climbers can't hold on to a wall forever).

Reply:
it should go in `main`. Since it's here, the function is being called per player meaning the more players there are on the server the faster the shulkers die... if there are 6 or more players online shulkers die immediately on the next clock cycle.

Reply:
If this is moved to `main` every command uses `at @s`, so the `at @s` should be moved to `main` when calling `player_main` and these commands should just use `if entity`